src/mscorlib/src/System/Span.cs

```diff
@@ -289,6 +289,25 @@ public void Set(ReadOnlySpan<T> values)
 
             SpanHelper.CopyTo<T>(ref GetRawPointer(), ref values.GetRawPointer(), values.Length);
         }
+
+        /// <summary>
+        /// Fills the contents of this span with the value.
+        /// </summary>
+        /// <param name="value">The value to fill the span content with.</param>
+        public void Fill(T value)
+        {
+            if (_length > 0)
+                SpanHelper.Fill<T>(ref GetRawPointer(), value, _length);
+        }
+
+        /// <summary>
+        /// Fills the contents of this span with the default value.
+        /// </summary>
+        public void Clear()
+        {
+            if (_length > 0)
+                SpanHelper.Fill<T>(ref GetRawPointer(), default(T), _length);
```

Comment:
I don't think that `Clear` should use `Fill`. `Clear` can be implemented more efficiently as it doesn't care about the size of `T` nor about references. It simply needs to fill the memory with 0.